contrib/ros/envs/sawyer/pick_and_place_env.py

```diff
@@ -2,51 +2,156 @@
 Pick-and-place task for the sawyer robot
 """
 
+from geometry_msgs.msg import Pose, Point
 import numpy as np
 
 from rllab.core.serializable import Serializable
+from rllab.envs.base import Step
+from rllab.spaces import Box
 
-from contrib.ros.envs import sawyer_env
+from contrib.ros.envs.ros_env import RosEnv
 from contrib.ros.robots.sawyer import Sawyer
+from contrib.ros.util.common import rate_limited
 
-INITIAL_ROBOT_JOINT_POS = {
-    'right_j0': -0.041662954890248294,
-    'right_j1': -1.0258291091425074,
-    'right_j2': 0.0293680414401436,
-    'right_j3': 2.17518162913313,
-    'right_j4': -0.06703022873354225,
-    'right_j5': 0.3968371433926965,
-    'right_j6': 1.7659649178699421,
-}
 
-
-class PickAndPlaceEnv(sawyer_env.SawyerEnv, Serializable):
+class PickAndPlaceEnv(RosEnv, Serializable):
     def __init__(self,
                  initial_goal,
                  task_obj_mgr,
                  sparse_reward=True,
                  simulated=False,
                  distance_threshold=0.05,
                  target_range=0.15,
-                 target_in_the_air=False):
+                 robot_control_mode='position'):
         Serializable.quick_init(self, locals())
 
         self._distance_threshold = distance_threshold
         self._target_range = target_range
         self._sparse_reward = sparse_reward
-        self._target_in_the_air = target_in_the_air
+        self.task_obj_mgr = task_obj_mgr
         self.initial_goal = initial_goal
         self.goal = self.initial_goal.copy()
+        self.simulated = simulated
+
+        self._sawyer = Sawyer(
+            simulated=simulated, control_mode=robot_control_mode)
+
+        RosEnv.__init__(self, simulated=simulated)
+
+    def _initial_setup(self):
+        self._sawyer.reset()
+
+        if self.simulated:
+            # Generate the world
+            # Load the model
+            for obj in self.task_obj_mgr.objects:
+                self.gazebo.load_gazebo_model(obj)
+        else:
+            # TODO(gh/8: Sawyer runtime support)
+            pass
+
+    def shutdown(self):
+        if self.simulated:
+            # delete model
+            for obj in self.task_obj_mgr.objects:
+                self.gazebo.delete_gazebo_model(obj.name)
+        else:
+            # TODO(gh/8: Sawyer runtime support)
+            pass
+
+    def reset(self):
+        """
+        Resets the state of the environment, returning an initial observation.
+        Outputs
+        -------
+        observation : the initial observation of the space. (Initial reward is assumed to be 0.)
+        """
+        self._sawyer.reset()
+
+        self.goal = self.sample_goal()
+
+        if self.simulated:
```

Comment:
With few exceptions, all of these switches `if self.simulated` switches should be gone entirely. You are lacking a good simulated vs real abstraction and it shows.

When I make an environment simulated vs real, two things change: the robot and the world (TaskObjectManager). Nothing about how I define the task should change. Therefore, with some occasional exceptions because of weird timing/system behavior, the only place is should see `if self.simulated` is in a robot or TaskObjectManager.